src/websocket/server.cc

```diff
@@ -27,11 +27,28 @@
 #include <seastar/core/scattered_message.hh>
 #include <seastar/core/byteorder.hh>
 #include <seastar/http/request.hh>
+#ifdef SEASTAR_WITH_TLS_OSSL
+#include <openssl/evp.h>
+#else
 #include <gnutls/crypto.h>
 #include <gnutls/gnutls.h>
+#endif
 
 namespace seastar::experimental::websocket {
 
+#ifdef SEASTAR_WITH_TLS_OSSL
+template<typename T, auto fn>
+struct ossl_deleter {
+    void operator()(T* ptr) { fn(ptr); }
+};
+
+template<typename T, auto fn>
+using ossl_handle = std::unique_ptr<T, ossl_deleter<T, fn>>;
+
+using evp_md_ptr = ossl_handle<EVP_MD, EVP_MD_free>;
+using evp_md_ctx_ptr = ossl_handle<EVP_MD_CTX, EVP_MD_CTX_free>;
```

Comment:
Could avoid this and just use a defer construct like whats going on in the existing tls stuff

Reply:
yeah... I guess I could do that.

Reply:
up to the author though !